CONTRIBUTING.md

```diff
@@ -76,15 +76,50 @@ To address feedback from code review, implement changes and push to your feature
 ### Code style and formatting
 The Lingua Franca compiler is implemented in Java and Kotlin. The overarching advice is to use each language's most widely used idioms and conventions, which are fortunately very similar. The code base is shipped with a [Spotless](https://github.com/diffplug/spotless) configuration to check and enforce style compliance. Lingua Franca code (e.g., tests) in this repository is also automatically formatted via Spotless.
 
+
+#### Formatting
+
 - To check that modified files are formatted correctly, run:
+
 ```
 ./gradlew spotlessCheck
 ```
+
 - To apply the changes recommended by the formatter, run:
+
 ```
 ./gradlew spotlessApply
 ```
 
+_Java code formatting in IntelliJ_
+
+There is an [IntelliJ plugin](https://plugins.jetbrains.com/plugin/8527-google-java-format) for enforcing Google's Java
+style guide. This plugin can be enabled by going to `File` > `Settings...` > `Plugins` > `MarketPlace` and searching for
+`google-java-format`. Do not forget to update your VM options as
+explained [here](https://github.com/google/google-java-format/blob/master/README.md#intellij-android-studio-and-other-jetbrains-ides)
+and do not forget to enable the plugin by going
+to `File` > `Settings...` > `google-java-format Settings` > `Enable google-java-format`.
+
+You can enable formatting on save by going to `File` > `Settings...` > `Tools` > `Actions on Save` > `Reformat Code`. If
+you do this, you may choose to change the files formatted on save from "all file types" to just Java in order to avoid
+unnecessary conflicts.
+
+_Lingua Franca code formatting in VS Code_
+
+Formatting is provided by the Lingua Franca VS Code extension.
+
+You can enable formatting on save by going to `Settings` and searching for "format" and checking the box
+`Editor: Format on Save`. This can be either a user-level or workspace-level setting.
+
+_Checking using Git hooks_
+
+If you prefer not to use formatting on save but would still like to avoid committing unformatted code, you can add a
+file `.git/hooks/pre-commit` with the contents:
+
+```
+./gradlew spotlessCheck
```

Comment:
Maybe better to suggest `spotlessApply` here?

Reply:
I have tried that, but the problem is that the changes that come from `spotlessApply` will not be staged. You could run `git add -a` in the hook, of course, but that would be dangerous because people would end up accidentally committing changes that they did not mean to commit.

Reply:
Ah, I see. Does [this](https://github.com/jguttman94/pre-commit-gradle) solve that?

Reply:
I do not know. If anyone wants to investigate that, we can update `CONTRIBUTING.md`.

Reply:
Maybe @axmmisaka can have a look. Marking this as resolved for now.

Reply:
We could do something like this (inspired [here](https://medium.com/@mmessell/apply-spotless-formatting-with-git-pre-commit-hook-1c484ea68c34)):

```bash
f = $(git diff --name-only --cached --diff-filter=AM)
for file in $f; do
  if test -f "$file"; then # this if might be redundant, as we applied filter in `git diff`
    git add $file
  fi
done
git commit
```